### PR TITLE
InvalidTargetDataCheck: if y is None, return data check error instead of raising exception

### LINKED ISSUE #1820: InvalidTargetDataCheck: if y is None, raise data check error
Original comment from @dsherry: https://github.com/alteryx/evalml/issues/1726#issuecomment-772647856

However, it feels weird to me for a data check that expects a target / checks for invalid target to be okay with not being passed a target. RE original comment, perhaps for unsupervised problems we shouldn't run that data check, rather than update the data check itself.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1820

### CODE REVIEW

**Review by jeremyliweishih — APPROVED** [review]
nice nice

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by freddyaboulton — APPROVED** [review]
Thanks @angela97lin !!

File: evalml/pipelines/pipeline_base_meta.py [hunk 1]
```diff
@@ -31,3 +31,28 @@ def _check_for_fit(self, X=None, y=None, objective=None):
...
+    """Metaclass that overrides creating a new pipeline by wrapping methods with validators and setters"""
+
+    @classmethod
+    def check_for_fit(cls, method):
+        """`check_for_fit` wraps a method that validates if `self._is_fitted` is `True`.
+            It raises an exception if `False` and calls and returns the wrapped method if `True`.
+        """
+        @wraps(method)
+        def _check_for_fit(self, X=None, y=None, objective=None):
+            klass = type(self).__name__
```
Comment (chukarsten) [inline evalml/pipelines/pipeline_base_meta.py:45]: Hahaha, really?

**Review by chukarsten — APPROVED** [review]
Angela, looks good.

File: evalml/pipelines/pipeline_base_meta.py (hunk 1 above)
Comment (angela97lin) [reply evalml/pipelines/pipeline_base_meta.py]: this is what I get for trying to do work mindlessly between meetings 😭

**Pull request merged**